app/cdap/components/PushdownConfig/index.tsx

```diff
@@ -0,0 +1,194 @@
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { MyPipelineApi } from 'api/pipeline';
+import { fetchPluginWidget } from 'services/PluginUtilities';
+import ConfigurationGroup from 'components/ConfigurationGroup';
+import { objectQuery } from 'services/helpers';
+import ToggleSwitch from 'components/ToggleSwitch';
+import LoadingSVG from 'components/LoadingSVG';
+import VersionStore from 'services/VersionStore';
+
+const useStyles = makeStyles({
+  container: {
+    left: 0,
+    maxWidth: '100%',
+    width: '100%',
+    height: '100%',
+    paddingBottom: '50px',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'stretch',
+    justifyContent: 'stretch',
+  },
+  toggleContainer: {
+    maxWidth: '25%',
+    paddingTop: '10px',
+    paddingBottom: '10px',
+  },
+  inner: {
+    overflowY: 'scroll',
+  },
+});
+
+function defaultPushdown(version) {
+  return {
+    // TODO(lahwran): when more pushdown plugins are supported, will need to allow user to select which one to use and use its info here
+    plugin: {
+      name: 'BigQueryPushdownEngine',
+      label: 'BigQueryPushdown',
+      type: 'sqlengine',
+      artifact: {
+        name: 'google-cloud',
+        version,
+        scope: 'SYSTEM',
+      },
+      properties: {},
+    },
+  };
+}
+
+function fetchPluginInfo() {
+  const version = VersionStore.getState().version;
+  const defaults = defaultPushdown(null);
+  const pluginParams = {
+    namespace: getCurrentNamespace(),
+    parentArtifact: 'cdap-data-pipeline',
+    version,
+    extension: defaults.plugin.type,
+    pluginName: defaults.plugin.name,
+    scope: 'SYSTEM',
+    artifactName: defaults.plugin.artifact.name,
+    artifactScope: defaults.plugin.artifact.scope,
+    limit: 1,
+    order: 'DESC',
+  };
+
+  return MyPipelineApi.getPluginProperties(pluginParams).map(([res]) => {
+    return res;
+  });
+}
+interface IPushdownConfig {
+  enabled: boolean;
+  transformationPushdown: null | {
+    plugin: null | object;
+  };
+}
+
+interface IPushdownProps {
+  value: IPushdownConfig;
+  onValueChange: (value: IPushdownConfig) => void;
+}
+
+export default function PushdownConfig({ value, onValueChange }: IPushdownProps) {
+  const [loading, setLoading] = useState(true);
+  const [pluginInfo, setPluginInfo] = useState(null);
+  const [pluginWidget, setPluginWidget] = useState(null);
+  const pluginProperties = objectQuery(pluginInfo, 'properties') || {};
+  const { enabled, transformationPushdown } = value;
+
+  const plugin = objectQuery(transformationPushdown, 'plugin');
+  const valueProperties = objectQuery(plugin, 'properties') || {};
+
+  const classes = useStyles();
+  const onChange = (newProps) => {
+    const newTransformationPushdown = {
+      plugin: {
+        ...plugin,
+        properties: {
+          ...valueProperties,
```

Comment:
this should just be `...newProps`, the onChange argument will get the entire value

Reply:
Ah, good to know